py/server/deephaven/table.py

```diff
@@ -55,13 +60,279 @@
 _JScriptSessionQueryScope = jpy.get_type("io.deephaven.engine.util.AbstractScriptSession$ScriptSessionQueryScope")
 _JPythonScriptSession = jpy.get_type("io.deephaven.integrations.python.PythonDeephavenSession")
 
+# Rollup Table and Tree Table
+_JRollupTable = jpy.get_type("io.deephaven.engine.table.hierarchical.RollupTable")
+_JTreeTable = jpy.get_type("io.deephaven.engine.table.hierarchical.TreeTable")
+_JRollupTableNodeOperationsRecorder = jpy.get_type(
+    "io.deephaven.engine.table.hierarchical.RollupTable$NodeOperationsRecorder")
+_JTreeTableNodeOperationsRecorder = jpy.get_type(
+    "io.deephaven.engine.table.hierarchical.TreeTable$NodeOperationsRecorder")
+_JNodeType = jpy.get_type("io.deephaven.engine.table.hierarchical.RollupTable$NodeType")
+_JFormatOperationsRecorder = jpy.get_type("io.deephaven.engine.table.hierarchical.FormatOperationsRecorder")
+_JSortOperationsRecorder = jpy.get_type("io.deephaven.engine.table.hierarchical.SortOperationsRecorder")
+_JFilterOperationsRecorder = jpy.get_type("io.deephaven.engine.table.hierarchical.FilterOperationsRecorder")
+
 # For unittest vectorization
 _test_vectorization = False
 _vectorized_count = 0
 
-# Rollup Table and Tree Table
-_JRollupTable = jpy.get_type("io.deephaven.engine.table.hierarchical.RollupTable")
-_JTreeTable = jpy.get_type("io.deephaven.engine.table.hierarchical.TreeTable")
+
+class NodeType(Enum):
+    """An enum of node types for RollupTable"""
+    AGGREGATED = _JNodeType.Aggregated
+    """Nodes at one of the aggregated (rolled up) levels in the RollupTable. These nodes have column names and types 
+    that result from applying aggregations on the source table of the RollupTable."""
+    CONSTITUENT = _JNodeType.Constituent
+    """Nodes at the leaf level when meth:`~deephaven.table.Table.rollup` method is called with 
+    include_constituent=True. These nodes have column names and types from the source table of the RollupTable."""
```

Comment:
```suggestion
    """Nodes at the leaf level when meth:`~deephaven.table.Table.rollup` method is called with 
    include_constituent=True. The constituent table is the lowest level in a rollup table. These nodes have column names and types from the source table of the RollupTable."""
```

Reply:
I hope this and the above suggestion help explain a bit, since hierarchy is important to understand in a rollup table.